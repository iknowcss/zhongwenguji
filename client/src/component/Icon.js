import React from 'react';
import cx from 'classnames';
import style from './Icon.module.scss';
import undoSvg from './undo.svg';
import searchSvg from './search.svg';
import swipeLeftSvg from './swipeLeft.svg';
import swipeRightSvg from './swipeRight.svg';
import keyboardArrowLeftSvg from './keyboardArrowLeft.svg';
import keyboardArrowRightSvg from './keyboardArrowRight.svg';

const Icon = ({ className, viewBox, href, size = 'small' }) => (
  <span className={cx(style.span, style[`span--${size}`], className)}>
    <svg viewBox={viewBox} className={style.svg}>
      <use xlinkHref={href} />
    </svg>
  </span>
);

const iconClass = ({ baseClassName, href, viewBox }) => ({ className, ...props }) => (
  <Icon
    className={cx(baseClassName, className)}
    viewBox={viewBox}
    href={href}
    {...props}
  />
);

export const UndoIcon = iconClass({
  href: `${undoSvg}#main`,
  baseClassName: style.undoIcon,
  viewBox: '0 0 511.63 511.631'
});

export const SearchIcon = iconClass({
  href: `${searchSvg}#main`,
  baseClassName: style.searchIcon,
  viewBox: '0 0 250.313 250.313'
});

export const SwipeLeftIcon = iconClass({
  href: `${swipeLeftSvg}#main`,
  baseClassName: style.swipeLeftIcon,
  viewBox: '0 0 397.232 397.232'
});

export const SwipeRightIcon = iconClass({
  href: `${swipeRightSvg}#main`,
  baseClassName: style.swipeRightIcon,
  viewBox: '0 0 397.233 397.233'
});

export const KeyboardArrowLeftIcon = iconClass({
  href: `${keyboardArrowLeftSvg}#main`,
  baseClassName: style.keyboardArrowLeftIcon,
  viewBox: '0 0 377.338 377.338'
});

export const KeyboardArrowRightIcon = iconClass({
  href: `${keyboardArrowRightSvg}#main`,
  baseClassName: style.keyboardArrowRightIcon,
  viewBox: '0 0 419.901 419.901'
});

export const LoadingIcon = ({ className, size = 'small' }) => (
  <span className={cx(style.span, style[`span--${size}`], style.loadingIcon, className)}>
    <span>加载</span>
    <span />
    <span />
    <span />
    <span />
  </span>
);
