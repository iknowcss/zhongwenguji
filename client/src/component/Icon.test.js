import React from 'react';
import Renderer from 'react-test-renderer';
import { LoadingIcon } from './Icon';

describe('Icon', () => {
  it('renders LoadingIcon with default size', () => {
    const component = Renderer.create(<LoadingIcon />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders LoadingIcon with medium size', () => {
    const component = Renderer.create(<LoadingIcon size="medium" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
