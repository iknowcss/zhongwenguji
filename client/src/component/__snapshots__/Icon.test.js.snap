// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders LoadingIcon with default size 1`] = `
<span
  className="span span--small loadingIcon"
>
  <span>
    加载
  </span>
  <span />
  <span />
  <span />
  <span />
</span>
`;

exports[`Icon renders LoadingIcon with medium size 1`] = `
<span
  className="span span--medium loadingIcon"
>
  <span>
    加载
  </span>
  <span />
  <span />
  <span />
  <span />
</span>
`;
